### PR TITLE
refactor: deprecate supply royalty

### PR DESCRIPTION
**Pull request by kulkarohan**
before margin: 0.117 kb
~available margin: 0.839 kb~
available margin: 0.576 kb

### CODE REVIEW

**Comment by kulkarohan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #267** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/267" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/267?utm_source=stack-comment).

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 1]
```diff
@@ -553,63 +554,8 @@ contract ZoraCreator1155Impl is
...
-    /// @param to to mint to
-    /// @param id token id to mint
-    /// @param amount of tokens to mint
-    /// @param data as specified by 1155 standard
-    function _mint(address to, uint256 id, uint256 amount, bytes memory data) internal virtual override {
-        uint256 supplyRoyaltyMints = _handleSupplyRoyalty(id, amount, data);
-        _requireCanMintQuantity(id, amount + supplyRoyaltyMints);
-
-        super._mint(to, id, amount, data);
-        tokens[id].totalMinted += amount + supplyRoyaltyMints;
```
Comment (oveddan) [inline packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:595]: do we still want to track `totalMinted` ? are we using that anywhere?

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 2]
```diff
@@ -388,7 +388,8 @@ contract ZoraCreator1155Impl is
                 _requireAdminOrRole(msg.sender, tokenIds[i], PERMISSION_BIT_MINTER);
             }
         }
-        _mintBatch(recipient, tokenIds, quantities, data);
+
+        super._mintBatch(recipient, tokenIds, quantities, data);
```
Comment (oveddan) [inline packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:392]: nice!

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 3]
```diff
@@ -553,63 +554,8 @@ contract ZoraCreator1155Impl is
         return super.supportsInterface(interfaceId) || interfaceId == type(IZoraCreator1155).interfaceId || ERC1155Upgradeable.supportsInterface(interfaceId);
     }
 
-    function _handleSupplyRoyalty(uint256 tokenId, uint256 mintAmount, bytes memory data) internal returns (uint256 totalRoyaltyMints) {
```
Comment (oveddan) [inline packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:556]: we're ok removing this?

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol (hunk 1 above)
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:595]: ah good catch - @iainnash wdyt?

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol (hunk 3 above)
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:556]: yup got the approval!

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol (hunk 1 above)
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:595]: i just checked our frontend repo and they're using it - i'll add that back

Comment (iainnash) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:595]: yep we want to track this still.

**Review by iainnash — CHANGES_REQUESTED** [review]
looks good other than dan's catch for totalMinted

**Comment by kulkarohan** [comment]
sooo i forgot about the `_requireCanMintQuantity` check also

**Comment by kulkarohan** [comment]
adding back the overrides 😭

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 4]
```diff
@@ -553,61 +554,36 @@ contract ZoraCreator1155Impl is
...
-    /// @notice Mint function that 1) checks quantity and 2) handles supply royalty 3) keeps track of allowed tokens
+    /// @notice Mint function that 1) checks quantity 2) keeps track of allowed tokens
     /// @param to to mint to
     /// @param id token id to mint
     /// @param amount of tokens to mint
     /// @param data as specified by 1155 standard
     function _mint(address to, uint256 id, uint256 amount, bytes memory data) internal virtual override {
-        uint256 supplyRoyaltyMints = _handleSupplyRoyalty(id, amount, data);
-        _requireCanMintQuantity(id, amount + supplyRoyaltyMints);
+        _requireCanMintQuantity(id, amount);
```
Comment (oveddan) [inline packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:565]: we need this check?

**Comment by iainnash** [comment]
yes for limited size editions

**Review by oveddan — CHANGES_REQUESTED** [review]
looks great! just please add a `changeset` (with `npx changeset` from the root of the project) and a `minor` change then lgtm

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol (hunk 4 above)
Comment (iainnash) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:565]: yes for limited supply nfts.

File: packages/1155-contracts/src/royalties/CreatorRoyaltiesControl.sol [hunk 5]
```diff
@@ -33,8 +33,8 @@ abstract contract CreatorRoyaltiesControl is CreatorRoyaltiesStorageV1, SharedBa
     }
 
     function _updateRoyalties(uint256 tokenId, RoyaltyConfiguration memory configuration) internal {
```
Comment (oveddan) [inline packages/1155-contracts/src/royalties/CreatorRoyaltiesControl.sol:35]: how about just letting this take in a royalty recipient address, and removing the concept of `RoyaltyConfiguration` ?

**Pull request merged**